installer/pkg/config/parser.go

```diff
@@ -42,6 +36,126 @@ func ParseConfig(data []byte) (*Cluster, error) {
 	return &cluster, nil
 }
 
+func parseInstallConfig(data []byte, cluster *Cluster) (err error) {
+	installConfig := &types.InstallConfig{}
```

Comment:
It would be nice if the defaulting, and validation were both extracted separately here such that external callers could vendor and use that code. We're still trying to determine if we can re-use your InstallConfig or we have to use our own type and generate InstallConfig, but it might be good to do this regardless for futureproofing.

Reply:
> It would be nice if the defaulting, and validation were both extracted separately here such that external callers could vendor and use that code.

New external consumers should be using `pkg/types` directly; everything under `installer/pkg` should be considered deprecated.  Having a validator for these types under `pkg/` makes sense (currently `pkg/asset/installconfig` has user-input-time validation, but that's not useful for downstream Go consumers reading this from the cluster, e.g. openshift/machine-config-operator#47).  And yeah, we'll probably need to expose the defaulting for tools that are monitoring InstallConfig to pick up post-install changes.  But that will also be a `pkg/` change, and this PR is trying to focus on getting support in the legacy `installer/pkg/config` (which has its own validation and defaulting approach).  Can I punt the new APIs to follow-up work?

Reply:
Totally fine with me.